dietpi/dietpi-software

```diff
@@ -3526,31 +3526,20 @@ _EOF_
 			if [[ $version == 'Stable' ]]
 			then
 				Download_Install "$(curl -sSfL 'https://api.github.com/repos/ravenclaw900/dietpi-dashboard/releases/latest' | mawk -F\" "/\"browser_download_url\": \".*dietpi-dashboard-$G_HW_ARCH_NAME\"/{print \$4}")" /opt/dietpi-dashboard/dietpi-dashboard
+				[[ -f '/opt/dietpi-dashboard/config.toml' ]] || Download_Install "$(curl -sSfL 'https://api.github.com/repos/ravenclaw900/dietpi-dashboard/releases/latest' | mawk -F\" "/\"browser_download_url\": \".*config.toml\"/{print \$4}")" /opt/dietpi-dashboard/config.toml
 
 			elif [[ $version == 'Nightly' ]]
 			then
 				no_check_url=1 Download_Install "https://nightly.link/ravenclaw900/DietPi-Dashboard/workflows/push-build/main/dietpi-dashboard-$G_HW_ARCH_NAME.zip" /opt/dietpi-dashboard
+				[[ -f '/opt/dietpi-dashboard/config.toml' ]] || Download_Install "https://raw.githubusercontent.com/ravenclaw900/DietPi-Dashboard/main/config.toml" /opt/dietpi-dashboard/config.toml
 			fi
 
 			G_EXEC chmod +x /opt/dietpi-dashboard/dietpi-dashboard
 
-			[[ -f '/opt/dietpi-dashboard/config.toml' ]] || cat << _EOF_ > /opt/dietpi-dashboard/config.toml
-# TCP network port
-# - Default: 8088
-#port = 8088
-
-# TLS for HTTPS
-# - Default: false
-#tls = false
-#cert = "/path/to/cert"
-#key = "/path/to/key"
-
-# Password protection
-# - Default: false
-pass = true
-hash = "$(echo -n "$GLOBAL_PW" | sha512sum | mawk '{print $1}')"
-secret = "$(openssl rand -hex 32)"
-_EOF_
+			G_CONFIG_INJECT '#pass[[:blank:]]=' 'pass = true' /opt/dietpi-dashboard/config.toml
+			G_CONFIG_INJECT '#hash[[:blank:]]=' "hash = \"$(echo -n "$GLOBAL_PW" | sha512sum | mawk '{print $1}')\"" /opt/dietpi-dashboard/config.toml
+			G_CONFIG_INJECT '#secret[[:blank:]]=' "secret = \"$(openssl rand -hex 32)\"" /opt/dietpi-dashboard/config.toml
```

Comment:
We must not reset the password and the secret on a reinstall. What about defining the URL for the config based on the choice and then do the download and config changes in a separate `[[ ! -f '/opt/dietpi-dashboard/config.toml' ]]` block below?